src/js/components/projects/detail.tsx

```diff
@@ -321,25 +328,44 @@ const ProjectDetail = (
                 />
               </div>
             )}
-            {hasEpics && (
-              <>
-                <EpicTable epics={epics.epics} projectSlug={project.slug} />
-                {epics.next ? (
-                  <div className="slds-m-top_large">
-                    <Button
-                      label={
-                        fetchingEpics ? (
-                          <LabelWithSpinner />
-                        ) : (
-                          i18n.t('Load More')
-                        )
-                      }
-                      onClick={fetchMoreEpics}
-                    />
-                  </div>
-                ) : null}
-              </>
-            )}
+            <Tabs variant="scoped" id="tabs-example-scoped">
+              <TabsPanel label={i18n.t('Epics')}>
+                {hasEpics && (
+                  <>
+                    <EpicTable epics={epics.epics} projectSlug={project.slug} />
+                    {epics.next ? (
+                      <div className="slds-m-top_large">
+                        <Button
+                          label={
+                            fetchingEpics ? (
+                              <LabelWithSpinner />
+                            ) : (
+                              i18n.t('Load More')
+                            )
+                          }
+                          onClick={fetchMoreEpics}
+                        />
+                      </div>
+                    ) : null}
+                  </>
+                )}
+              </TabsPanel>
+              <TabsPanel label={i18n.t('Tasks')}>
+                {hasTasks && (
+                  <TasksTableComponent
+                    projectId={project.id}
+                    projectSlug={project.slug}
+                    tasks={tasks as Task[]}
+                    githubUsers={project.github_users}
+                    canAssign={project.has_push_permission}
+                    isRefreshingUsers={Boolean(
+                      project.currently_refreshing_gh_users,
+                    )}
+                    assignUserAction={assignUser}
+                  />
```

Comment:
@jerivas We're running into an issue with this new "All Tasks" table, because some of the info in the table relies on having access to the `task.epic`, but because of the context here (project-detail page) and pagination we are not (currently) guaranteed to have fetched the parent Epic for _each task_ at this point. Specifically, the data we're missing is:

- `epic.slug` (needed to generate the URL for the Epic)
- `epic.name` (to display in the table)
- `epic.github_users` (to show in the modal when assigning a user directly to a Task dev/test role)

There would be various options here for the first two items, and I'm curious what you think:

1. Add the required info to the serialized Task
2. Add user-facing URLs to serialized objects (e.g. `task.app_url` and `epic.app_url`) -- we would still need `epic.name` added too
3. Make an extra API call to get the parent Epic for each Task

Finally, there are also various options for the `epic.github_users` problem:

1. Just don't display `Epic Collaborators` in this case, and instead display all GitHub users for the Project when you go to assign a user to a task role from this view. (I'd want @SondraE's input on this approach too)
2. Add the required `epic.github_users` data to the serialized Task
3. Make an extra API call to get the parent Epic for each Task (probably deferred until the AssignUserModal is opened)

Any thoughts? I'm happy to jump on a call if that's simpler.

Reply:
For the Epic slug and name we could get a list all Epics referenced by a particular page of Tasks, get them all in one query, and either nest them in the task, or provide them as a separate key of the payload (similar to the multi-model responses we do for `/related/` endpoints in GSM)

For collaborators I would definitely go with option 3 (defer the parent Epic until the modal is opened)

Reply:
Okay, I think that makes sense. A separate key in the payload sounds cleaner. Though if we're already doing that, I'd be tempted to just include `github_users` in that as well -- it's a pretty simple list of strings I think?

Reply:
You're right, it's not really expensive to calculate. We should be able to get it in there.

Let me do some profiling to actually test if separate keys are better than nesting. Perhaps serializing the Epic in place is good enough if we use `select_related` correctly

Reply:
@jgerigmeyer I just noticed Tasks are not paginated. I think that's something we might want to do in this PR?

Reply:
@jerivas Yes, that should happen in this PR too. Initially I was thinking maybe we only paginate this new version of the list, but now I'm thinking we might as well do it everywhere.

Reply:
Alright, b272e3a61f231910c34eca030d11595868af5435 added pagination and nested epics to each `Task`. While profiling the changes I noticed we were getting a LOT of queries for `TaskSlug` and `EpicSlug`, and turns out it has to do with the implementation in `sfdo-template-helpers`. I opened a [pull request](https://github.com/SFDO-Tooling/sfdo-template-helpers/pull/41) that when merged will significantly reduce the number of queries in Task and Epic list endpoints.

Since we are now paginating both lists I'm not too worried about performance, but I created a [card](https://trello.com/c/UnY0CXxa/300-profile-endpoints-that-use-slugs) to keep an eye on it.